imblearn/over_sampling/_smote/base.py

```diff
@@ -786,7 +786,7 @@ def _make_samples(self, X_class, klass, y_dtype, nn_indices, n_samples):
         # where for each feature individually, each category generated is the
         # most common category
         X_new = np.squeeze(
-            stats.mode(X_class[nn_indices[samples_indices]], axis=1).mode, axis=1
+            _mode(X_class[nn_indices[samples_indices]], axis=1).mode, axis=1
         )
```

Comment:
When `scipy==1.9.0` becomes the minimum version, these lines can also be simplified:

```python
from scipy.stats import mode

X_new = mode(X_class[nn_indices[samples_indices]], axis=1, keepdims=False).mode
```